src/model/model_evaluator.h

```diff
@@ -32,6 +32,7 @@ class model_evaluator {
     struct imp;
     imp *  m_imp;
 public:
+    model_evaluator(model_evaluator & mev);
```

Comment:
Not sure about why you add this constructor. 
It breaks some assumptions with m_imp being unique.

Reply:
indeed it doesn't required